src/connection_manager.rs

```diff
@@ -590,6 +601,15 @@ fn handle_connect(mut state: WeakState, trans: transport::Transport,
     endpoint
 }
 
+fn increment_bs_count(state: &mut WeakState) -> io::Result<bool> {
+    lock_mut_state(state, move |s: &mut State| {
+        if s.bs_count.0 < s.bs_count.1 {
+            s.bs_count.0 = s.bs_count.0 + 1;
+            return Ok(true);
+        }
+        Ok(false)  // reached max bs connection
```

Comment:
Perhaps here reset the BS connections to allow further bootstrap calls to be made. May require a bootstrapping flag though to block and bootstrap connections whilst not bootstrapping (i.e. this one is finished, but a user may call it again).  Either that or when bootstrap is called reset the current bs connections to 0 !